src/Tools/dotnet-monitor/EndpointInfo/ServerEndpointInfoSource.cs

```diff
@@ -189,19 +206,29 @@ private async Task ResumeAndQueueEndpointInfo(ReversedDiagnosticsServer server,
 
                     foreach (IEndpointInfoSourceCallbacks callback in _callbacks)
                     {
-                        await callback.OnAddedEndpointInfoAsync(endpointInfo, token).ConfigureAwait(false);
+                        try
+                        {
+                            await callback.OnAddedEndpointInfoAsync(endpointInfo, token).ConfigureAwait(false);
+                        }
+                        catch (Exception ex)
+                        {
+                            exceptions.Add(ex);
+                        }
                     }
                 }
                 finally
                 {
                     _activeEndpointsSemaphore.Release();
                 }
             }
-            catch (Exception)
+            catch (Exception ex)
             {
-                server?.RemoveConnection(info.RuntimeInstanceCookie);
```

Comment:
I'm a bit torn on this approach. If we have a recoverable error, a retry from the app would be useful.